lib/server/services/shift/ShiftService.js

```diff
@@ -64,6 +65,29 @@ class ShiftService {
             typeSpecific: null,
         };
 
+        // Find all EoS reports from the last 16 hours and sort by most recent first
+        const pastShift = { start: new Date() - SHIFT_DURATION * 2, end: new Date() };
+        const prevShiftIssues = (await getShiftIssues(pastShift, { include: null, exclude: [] }))
+            .reverse()
+            .sort((a, b) => {
+                new Date(b.createdAt) - new Date(a.createdAt);
+            });
+
+        // Filter for the correct tags and EoS title and select the most recent report
+        const reportFormat = `End of shift report - ${shiftType} - `;
+        const tags = getEosReportTagsByType(shiftType);
+        const allEosReports = prevShiftIssues.filter((report) =>
+            tags.every((tag) => report.tags.map((x) => x.dataValues.text).includes(tag))
+            && report.title.startsWith(reportFormat));
+        const eosReport = allEosReports.length > 0 ? allEosReports[0] : { text: '' };
```

Comment:
As we discussed, here I would rather :
- Extract a function to format EoS report title : https://github.com/AliceO2Group/Bookkeeping/blob/main/lib/server/services/eosReport/formatEosReport.js#L34
- Use it to get the expected title of the previous EoS report